src/rules/no-space-in-emphasis.js

```diff
@@ -0,0 +1,284 @@
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @import { Heading, Node, Paragraph, TableCell } from "mdast";
+ * @import { MarkdownRuleDefinition } from "../types.js";
+ * @typedef {{marker: string, startIndex: number, endIndex: number}} EmphasisMarker
+ * @typedef {"spaceInEmphasis"} NoSpaceInEmphasisMessageIds
+ * @typedef {[{ includeStrikethrough?: boolean }]} NoSpaceInEmphasisOptions
+ * @typedef {MarkdownRuleDefinition<{ RuleOptions: NoSpaceInEmphasisOptions, MessageIds: NoSpaceInEmphasisMessageIds }>} NoSpaceInEmphasisRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const whitespacePattern = /[ \t]/u;
+
+/**
+ * Creates a marker pattern based on whether strikethrough should be included
+ * @param {boolean} includeStrikethrough Whether to include strikethrough markers
+ * @returns {RegExp} The marker pattern
+ */
+function createMarkerPattern(includeStrikethrough) {
+	if (includeStrikethrough) {
+		return /(?<!\\)(\*\*\*|\*\*|\*|___|__|_|~~|~)/gu;
+	}
+	return /(?<!\\)(\*\*\*|\*\*|\*|___|__|_)/gu;
+}
+
+/**
+ * Creates emphasis types set based on whether strikethrough should be included
+ * @param {boolean} includeStrikethrough Whether to include strikethrough types
+ * @returns {Set<string>} The emphasis types set
+ */
+function createEmphasisTypes(includeStrikethrough) {
+	const types = new Set(["emphasis", "strong"]);
+	if (includeStrikethrough) {
+		types.add("delete");
+	}
+	return types;
+}
+
+/**
+ * Finds all emphasis markers in the text
+ * @param {string} text The text to search
+ * @param {RegExp} pattern The marker pattern to use
+ * @returns {Array<EmphasisMarker>} Array of emphasis markers
+ */
+function findEmphasisMarkers(text, pattern) {
+	/** @type {Array<EmphasisMarker>} */
+	const markers = [];
+	/** @type {RegExpExecArray | null} */
+	let match;
+
+	while ((match = pattern.exec(text)) !== null) {
+		markers.push({
+			marker: match[1],
+			startIndex: match.index,
+			endIndex: match.index + match[1].length,
+		});
+	}
+
+	return markers;
+}
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoSpaceInEmphasisRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description: "Disallow spaces around emphasis markers",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-space-in-emphasis.md",
+		},
+
+		fixable: "whitespace",
+
+		messages: {
+			spaceInEmphasis: "Unexpected space around emphasis marker.",
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					includeStrikethrough: {
```

Comment:
```suggestion
					checkStrikethrough: {
```

I think `checkStrikethrough` is a more appropriate option name, since many of the rules use `check` for boolean option names.

Here are some references:

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/no-duplicate-headings.js#L64-L66

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/no-empty-definitions.js#L73-L75

https://github.com/eslint/markdown/blob/614ef3e0bf539655170f202467119fbbbf8963f6/src/rules/table-column-count.js#L44-L46